src/tools/runk/libcontainer/src/created_builder.rs

```diff
@@ -0,0 +1,150 @@
+        debug!(logger, "enter CreatedContainer::create_launcher {:?}", self);
+        let container = Container::load(&self.root, &self.id)?;
+
+        if container.state != ContainerState::Created {
+            return Err(anyhow!(
+                "cannot start a container in the {:?} state",
+                container.state
+            ));
+        }
+
+        let config = container.status.config.clone();
+
+        debug!(
+            logger,
+            "Prepare LinuxContainer for starting with config: {:?}", config
+        );
+        let runner = load_linux_container(&container, None, logger)?;
+
+        Ok(ContainerLauncher::new(
+            &self.id,
+            &container.status.bundle,
+            &self.root,
+            true,
+            runner,
+            None,
+        ))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::container::CONFIG_FILE_NAME;
+    use crate::init_builder::InitContainerBuilder;
+    use crate::utils::test_utils::*;
+    use rustjail::specconv::CreateOpts;
+    use slog::o;
+    use std::fs::{create_dir, File};
+    use std::path::Path;
+    use tempfile::tempdir;
+
+    #[test]
+    fn test_init_container_validate() {
+        let root = tempdir().unwrap();
+        let id = TEST_CONTAINER_ID.to_string();
+        let result = CreatedContainerBuilder::default()
+            .id(id)
+            .root(root.path().to_path_buf())
+            .build();
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_init_container_create_launcher() {
+        let logger = slog::Logger::root(slog::Discard, o!());
+        let root_dir = tempdir().unwrap();
+        let bundle_dir = tempdir().unwrap();
+        // create dummy rootfs
+        create_dir(bundle_dir.path().join(TEST_ROOTFS_PATH)).unwrap();
+        let config_file = bundle_dir.path().join(CONFIG_FILE_NAME);
+        let mut spec = create_dummy_spec();
+        let file = File::create(config_file).unwrap();
+        serde_json::to_writer(&file, &spec).unwrap();
+
+        spec.root.as_mut().unwrap().path = bundle_dir
+            .path()
+            .join(TEST_ROOTFS_PATH)
+            .to_string_lossy()
+            .to_string();
+        let test_data = TestContainerData {
+            // Since tests are executed concurrently, container_id must be unique in tests with cgroup.
+            // Or the cgroup directory may be removed by other tests in advance.
+            id: String::from("test_created_container_create_launcher"),
+            bundle: bundle_dir.path().to_path_buf(),
+            root: root_dir.into_path(),
+            console_socket: Some(PathBuf::from(TEST_CONSOLE_SOCKET_PATH)),
+            config: CreateOpts {
+                spec: Some(spec),
+                ..Default::default()
+            },
+            pid_file: Some(PathBuf::from(TEST_PID_FILE_PATH)),
+        };
+
+        let launcher = InitContainerBuilder::default()
```

Comment:
It should be `CreatedContainerBuilder`.

Reply:
Oh sorry, before implemting the test, I accidentaly created the PR. I created the test.

Reply:
@ManaSugi Thanks. Looks good to me now.